app/src/test/java/org/mozilla/tv/firefox/pinnedtile/PinnedTileViewModelTest.kt

```diff
@@ -0,0 +1,83 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pinnedtile
+
+import android.arch.lifecycle.Observer
+import org.junit.Assert
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.ArgumentMatchers
+import org.mockito.Mockito
+import org.robolectric.RobolectricTestRunner
+import org.robolectric.RuntimeEnvironment
+
+/**
+ * Unit tests for the [PinnedTileViewModel].
+ */
+@RunWith(RobolectricTestRunner::class)
+class PinnedTileViewModelTest {
+
+    private lateinit var pinnedTileViewModel: PinnedTileViewModel
+    private lateinit var pinnedTileRepo: PinnedTileRepo
+
+    @Before
+    fun setUp() {
+        pinnedTileRepo = PinnedTileRepo(RuntimeEnvironment.application)
+        pinnedTileViewModel = PinnedTileViewModel(pinnedTileRepo)
+    }
+
+    @Test
+    fun `WHEN repo emits a successful load THEN view model should emit a list of same number of tiles`() {
+        val observerSpy = Mockito.spy(Observer<List<PinnedTile>> {
+            Assert.assertTrue(it is List<PinnedTile>)
+            Assert.assertEquals(10, it!!.size)
+        })
+
+        pinnedTileViewModel.getTileList().observeForever(observerSpy)
+
+        Mockito.verify(observerSpy, Mockito.times(1)).onChanged(ArgumentMatchers.any())
+    }
+
+    @Test
+    fun `WHEN repo emits an updated list after add THEN view model should emit an updated list`() {
```

Comment:
(if the repo is mocked out / Robolectric removed, ignore this comment)

The name of this test doesn't fully reflect what it's testing.  Suggest updating to `WHEN a new pinned tile is added to repo THEN ...`, to reflect that this is testing how both classes integrate.

Reply:
This applies to the other tests as well.